server/server_test.go

```diff
@@ -228,18 +228,10 @@ func TestAcceptEncoding(t *testing.T) {
 	defer leaktest.AfterTest(t)
 	s := StartTestServer(t)
 	defer s.Stop()
-	// We can't use the standard test client. Create our own.
```

Comment:
Yeah, `DisableCompression` was (part of) the point of the test (I'm not sure why the proxy setting was there). We're no longer testing that we send uncompressed responses when no `Accept-Encoding` was given (go deletes the `Content-Encoding` header after decoding the body, so we can't cleanly tell whether it was ever compressed or not). 

We should still be using `DisableCompression` here, but it's so tricky to make a test that is sensitive to whether or not it was used that it's probably not worth it.